beacon-chain/db/kv/state.go

```diff
@@ -99,30 +114,112 @@ func (s *Store) SaveStates(ctx context.Context, states []iface.ReadOnlyBeaconSta
 		return errors.New("nil state")
 	}
 	multipleEncs := make([][]byte, len(states))
+	validatorsEntries := make(map[string][]byte)              // It's a map to make sure that you store only new validator entries.
```

Comment:
We dont need these unless the feature flag is enabled

Reply:
+1, nice to reduce the extra allocations

Reply:
This might complicate the code as you have to basically intialize it ouside the conditional as it wont compile otherwise. For the reduced complexity, initializing it and not using it should be fine.

Reply:
Yeah, had a discussion with @terencechain and decided to not cover it using the flag.

Reply:
But now i have out it under a new function `SaveStatesEfficient`. so, this issues does not hold good anymore.